chia/data_layer/data_layer.py

```diff
@@ -62,20 +63,26 @@ async def get_plugin_info(url: str) -> Tuple[str, Dict[str, Any]]:
         return url, {"error": f"ClientError: {e}"}
 
 
+@final
+@dataclasses.dataclass
 class DataLayer:
-    data_store: DataStore
     db_path: Path
     config: Dict[str, Any]
     root_path: Path
     log: logging.Logger
     wallet_rpc_init: Awaitable[WalletRpcClient]
-    state_changed_callback: Optional[StateChangedProtocol] = None
-    wallet_id: uint64
-    initialized: bool
     none_bytes: bytes32
     _server: Optional[ChiaServer]
     downloaders: List[str]
     uploaders: List[str]
+    server_files_location: Path
+    initialized: bool = False
+    _data_store: Optional[DataStore] = None
+    state_changed_callback: Optional[StateChangedProtocol] = None
+    _shut_down: bool = False
+    periodically_manage_data_task: Optional[asyncio.Task[None]] = None
+    _wallet_rpc: Optional[WalletRpcClient] = None
+    _subscription_lock: Optional[asyncio.Lock] = None
```

Comment:
```suggestion
    _subscription_lock: asyncio.Lock = field(default_factory=asyncio.Lock)
```

Could be ^ and you could drop the related property?

Reply:
I think its just you every time this kind of thing comes up feeling like it needs to be done in a async call for some reason because its from the async package :) But we have bunch of places where we instantiate async members (`Queue`, `Event` and even also `Lock` already) like this and there doesn't seem to be an issue with it. I don't really see a reason to add this extra optional/property thing but feel free to close this.

Reply:
https://github.com/chia-network/chia-blockchain/commit/96f7bbca1238ec3e0dbc1d5dfe6532b755646626

Reply:
The issue is with `asyncio.Queue` being created outside the loop.

https://github.com/Chia-Network/chia-blockchain/pull/12041/files

https://github.com/Chia-Network/chia-blockchain/issues/11980